app-backend/api/src/main/scala/analysis/Routes.scala

```diff
@@ -0,0 +1,97 @@
+package com.azavea.rf.api.analysis
+
+import com.azavea.rf.common._
+import com.azavea.rf.common.ast._
+import com.azavea.rf.datamodel._
+import com.azavea.rf.tool.ast.MapAlgebraAST
+import com.azavea.rf.tool.eval.PureInterpreter
+import com.azavea.rf.database.filter.Filterables._
+import com.azavea.maml.serve.InterpreterExceptionHandling
+import com.lonelyplanet.akka.http.extensions.PaginationDirectives
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import cats.implicits._
+import java.util.UUID
+
+import cats.effect.IO
+import com.azavea.rf.database.AnalysisDao
+import doobie.util.transactor.Transactor
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import doobie._
+import doobie.implicits._
+import doobie.Fragments.in
+import doobie.postgres._
+import doobie.postgres.implicits._
+
+
+
+trait AnalysisRoutes extends Authentication
+    with PaginationDirectives
+    with AnalysisQueryParametersDirective
+    with CommonHandlers
+    with UserErrorHandler
+    with InterpreterExceptionHandling {
+
+  val xa: Transactor[IO]
+
+  val analysisRoutes: Route = handleExceptions(userExceptionHandler) {
+    pathEndOrSingleSlash {
+      get { listAnalyses } ~
+      post { createAnalysis }
+    } ~
+    pathPrefix(JavaUUID) { analysisId =>
+      pathEndOrSingleSlash {
+        get { getAnalysis(analysisId) } ~
+        put { updateAnalysis(analysisId) } ~
+        delete { deleteAnalysis(analysisId) }
+      }
+    }
+  }
+
+  def listAnalyses: Route = authenticate { user =>
+    (withPagination & analysisQueryParameters) { (page, analysisParams) =>
+      complete {
+        AnalysisDao.query.filter(analysisParams).ownerFilter(user).page(page).transact(xa).unsafeToFuture
+      }
+    }
+  }
+
+  def createAnalysis: Route = authenticate { user =>
+    entity(as[Analysis.Create]) { newAnalysis =>
+      authorize(user.isInRootOrSameOrganizationAs(newAnalysis)) {
+        onSuccess(AnalysisDao.insertAnalysis(newAnalysis, user).transact(xa).unsafeToFuture) { analysis =>
+          handleExceptions(interpreterExceptionHandler) {
+            complete {
+              (StatusCodes.Created, analysis)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  def getAnalysis(analysisId: UUID): Route = authenticate { user =>
+    rejectEmptyResponse {
+      complete(AnalysisDao.query.filter(fr"id = ${analysisId}").selectOption.transact(xa).unsafeToFuture)
```

Comment:
I think we set it up so you can just `query.filter(analysisId)` here -- which saves us the trouble of debugging more strings later

Reply:
I am almost definitely guilty of ignoring that as well

Reply:
it's somewhat confusing when you consider that this can only be used on stuff where the id is a UUID, not others like TemplateVersions and Users :thinking: